compiler/sem/dfa.nim

```diff
@@ -709,6 +709,7 @@ proc isAnalysableFieldAccess*(orig: PNode; owner: PSym): bool =
   # wrong! Then the write barrier for the ``self.root`` assignment would
   # free the old data and all is lost! Lesson: Don't be too smart, trust the
   # lower level C++ optimizer to specialize this code.
+  # TODO: revisit if still relevant and how
```

Comment:
The documented problem is not related to C++. Also, for what it's worth, the new move analyser does exactly what is described to be a problem here and there are no issues so far.